### DISCUSSION TITLE
sqlalchemy How to dynamically add query conditions

### DISCUSSION THREAD

**Discussion by dxrvip** [sqlalchemy/sqlalchemy#10424] [Usage Questions] [answered]
sqlalchemy2.0

stmt = select(func.count(Post.id)).filter_by(**filters.dict(exclude={"create_time"}))
if filters.create_time:
    stmt.filter(cast(Post.create_time, DATE) == cast(filters.create_time, DATE))

print.:  SELECT count(post.id) AS count_1 
FROM post 
WHERE post.domain_id = :domain_id_1 AND post.promotion = :promotion_1

I cannot add query conditions dynamically
thank you for your reply

**Comment by dxrvip** [comment]
stmt = stmt.filter(cast(Post.create_time, DATE) == cast(filters.create_time, DATE))
Can it be added directly to filter_by() to check the date?

**Reply by CaselIT** [reply, marked as answer]
since you cast the column `cast(Post.create_time, DATE)`, no you can't use filter_by
